src/LocalDatastoreController.react-native.js

```diff
@@ -38,18 +39,52 @@ const LocalDatastoreController = {
   },
 
   async getAllContents(): Promise {
-    const LDS = {};
     const keys = await RNStorage.getAllKeys();
+    const batch = [];
     for (let i = 0; i < keys.length; i += 1) {
       const key = keys[i];
-      const value = await RNStorage.getItemAsync(key);
-      LDS[key] = JSON.parse(value);
+      if (isLocalDatastoreKey(key)) {
+        batch.push(key);
+      }
     }
+    const LDS = {};
+    const results = await RNStorage.multiGet(batch);
+    results.map((pair) => {
+      const [key, value] = pair;
+      try {
+        LDS[key] = JSON.parse(value);
+      } catch (error) {
+        LDS[key] = null;
+      }
+    });
     return Promise.resolve(LDS);
   },
 
-  clear(): Promise {
-    return RNStorage.clear();
+  async getRawStorage(): Promise {
+    const keys = await RNStorage.getAllKeys();
+    const storage = {};
+    const results = await RNStorage.multiGet(keys);
+    results.map((pair) => {
+      const [key, value] = pair;
+      storage[key] = value;
+    });
+    return Promise.resolve(storage);
```

Comment:
you can just return storage here as an async will auto wrap it.

Reply:
Thanks for clearing that up